### PR TITLE
Add shepp_logan_phantom, colorwheel, brick, grass, roughwall to the data module

### PR DESCRIPTION
**Pull request by hmaarrfk**
## Description

Not sure about any of their licenses. At least this is a step in the right direction.

## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: skimage/data/__init__.py [hunk 1]
```diff
@@ -98,6 +103,42 @@ def astronaut():
...
+    """
+    return load("grass.png")
+
+
+def rough_wall():
+    """Rough wall.
+
+    Returns
+    -------
+    rough_wall: (512, 512)
```
Comment (lagru) [inline skimage/data/__init__.py]:
```suggestion
    rough_wall: (512, 512) uint8 image
```

File: skimage/data/__init__.py [hunk 2]
```diff
@@ -98,6 +103,42 @@ def astronaut():
...
+    """
+    return load("brick.png")
+
+
+def grass():
+    """Grass.
+
+    Returns
+    -------
+    grass: (512, 512) uint8
```
Comment (lagru) [inline skimage/data/__init__.py]:
```suggestion
    grass: (512, 512) uint8 image
```

**Comment by hmaarrfk** [comment]
I do think we should replace old code with these functions. In my mind though, that is a secondary benefit. The primary benefit is adding these to the gallery with docstrings.

I can work on the few needed mods in the examples.

**Comment by stefanv** [comment]
W.r.t. licensing of Shepp-Logan phantom.  Instructions for generating the phantom is described in literature, so I don't think we have to worry about the copyright: it is essentially a mapping from what is written in that paper to an array.

See, e.g.,

https://jenda.hrach.eu/f2/cat-py/phantom.py
https://github.com/ismrmrd/ismrmrd-python-tools/blob/master/ismrmrdtools/simulation.py#L85

(In the latter link, you'll see instructions for a modified Shepp-Logan phantom.  We may want to check what that is, and whether we should include it.)

**Comment by hmaarrfk** [comment]
Yeah, that is what I thought. What about roughwall, grass and brick?

**Comment by stefanv** [comment]
Those are from http://sipi.usc.edu/database/database.php?volume=rotate, but looking at their licensing page, they are probably from P. Brodatz, "Textures: A Photographic Album for Artists and Designers", Dover Publications, New York, 1966 and "The copyright is probably retained by the author."

This has been investigated by [several](http://graphics.stanford.edu/projects/texture/faq/brodatz.html) [others](http://graphics.stanford.edu/projects/texture/faq/picard.txt) already.

The last link is the most interesting—it shows that the author's widow does *not* want these images distributed, and that we are in violation of their copyright.  We should therefore remove the images, and may consider replacing them with images from the last link.  The popular [VisTex](https://vismod.media.mit.edu/vismod/imagery/VisionTexture/vistex.html) database is not suitable for our purposes, since they restrict reuse to non-commercial applications only.

**Comment by hmaarrfk** [comment]
@stefanv I think appropriate images have now been found.

**Comment by hmaarrfk** [comment]
@stefanv I guess my original goal has now been overshadowed by the licensing concerns you brought up.

File: skimage/data/__init__.py [hunk 3]
```diff
@@ -154,8 +155,30 @@ def rough_wall():
...
+    return gravel()
 
+def gravel():
+    """Gravel
+
+    The original image was downloaded from
+    `CC0Textures <https://cc0textures.com/view.php?tex=Gravel04>`__ and
+    licensed under the Creative Commons CC0 License.
+
+    The downloaded image was then imported to GIMP and converted from color
```
Comment (stefanv) [inline skimage/data/__init__.py]: Rescaled using which interpolation?  Funny that you did this in GIMP, but maybe it was to get Lanczos which, very sadly, we still don't.

**Comment by stefanv** [comment]
And, thank you—perhaps not your original intent, but I think more valuable even!

File: skimage/data/__init__.py (hunk 3 above)
Comment (hmaarrfk) [reply skimage/data/__init__.py]: Yeah. The look of the image was more important than the actual science in these cases

Comment (hmaarrfk) [reply skimage/data/__init__.py]: Finally, the big advantage of using GIMP was that I wanted to make sure the images "looked" good.

**Comment by hmaarrfk** [comment]
@stefanv any more comments on this one?

**Comment by sciunto** [comment]
Original  issue: #3747

File: skimage/data/__init__.py (hunk 3 above)
Comment (stefanv) [reply skimage/data/__init__.py]: My concern remains the same: why include a dataset that we cannot reproduce? Why not write a small piece of Python to apply all the transformations and store that somewhere? I understand that you just wanted some good test data (success, there!), but I feel we should be able to retrace our steps.

Comment (hmaarrfk) [reply skimage/data/__init__.py]: Hmm, ok. I'll see what I can do there. I didn't understand you wanted reproducibility.

I'll see what I can do.

File: skimage/data/__init__.py [hunk 4]
```diff
@@ -366,6 +448,34 @@ def retina():
...
 
 
+def shepp_logan_phantom():
+    """Shepp Logan Phantom.
+
+    References
+    ----------
+    .. [1] L. A. Shepp and B. F. Logan, "The Fourier reconstruction of a head
+           section," in IEEE Transactions on Nuclear Science, vol. 21,
+           no. 3, pp. 21-43, June 1974. doi: 10.1109/TNS.1974.6499235
```
Comment (sciunto) [inline skimage/data/__init__.py]:
```suggestion
           no. 3, pp. 21-43, June 1974. :DOI:`10.1109/TNS.1974.6499235`
```

File: skimage/data/__init__.py (hunk 3 above)
Comment (hmaarrfk) [reply skimage/data/__init__.py]: Have we decided that jpeg give reproducible images? Most of the images are stored as jpeg, so even loading them will yield different results.

I get that we strive for reproducibility, but we aren't using these images to make definitive scientific conclusions. GIMP is the right tool for making a "nice image". scikit-image is a good tool for science. 

The goal of this data is to create a "nice image that doesn't violate copyright and that isn't too large". I think it is better to focus our attention span elsewhere.

**Comment by hmaarrfk** [comment]
<details> <summary> WIP Code for "reproducible" images </summary>

```
import sys; print(sys.version)
import platform; print(platform.platform())
import skimage; print(f"scikit-image version: {skimage.__version__}")
import numpy; print(f"numpy version: {numpy.__version__}")
import imageio; print(f"imageio version {imageio.__version__}")

from imageio import imread, imwrite
from skimage.transform import rescale
from skimage import img_as_ubyte
import requests
import zipfile

url = 'https://cdn.struffelproductions.com/file/cc0textures/Bricks25/%5B2K%5DBricks25.zip'
r = requests.get(url)
with open('[2K]Bricks25.zip', 'bw') as f:
    f.write(r.content)

with zipfile.ZipFile('[2K]Bricks25.zip') as z:
    z.extract('Bricks25_col.jpg')

bricks_orig = imread('Bricks25_col.jpg')
bricks = rescale(bricks_orig[:1024, :1024], (0.5, 0.5, 1))
imwrite('bricks.png', img_as_ubyte(bricks))
```
</details>

**Comment by stefanv** [comment]
Nice!!

**Comment by hmaarrfk** [comment]
Reproduction code is now in a comment in the function itself. It doesn't pollute the HTLM Documentation, but yet, when people want to, they still have easy access to it. This PR should definitely be squashed and merged.

File: skimage/data/__init__.py [hunk 5]
```diff
@@ -98,6 +104,189 @@ def astronaut():
...
+    rough_wall: (512, 512) uint8 image
+        Some rough wall.
+
+    """
+    from warnings import warn
+    warn("The rough_wall dataset has been removed due to licensing concerns."
+         "It has been replaced with the gravel dataset. This warning message"
+         "will be replaced with an error in scikit-image 0.17.", stacklevel=2)
+    return gravel()
+
```
Comment (sciunto) [inline skimage/data/__init__.py]: PEP8, missing blank line

File: skimage/data/__init__.py [hunk 6]
```diff
@@ -98,6 +104,189 @@ def astronaut():
...
+    ...     z.extract('Gravel04_col.jpg')
+
+    >>> from skimage.transform import resize
+    >>> gravel_orig = imageio.imread('Gravel04_col.jpg')
+    >>> gravel = resize(gravel_orig, (1024, 1024))
+    >>> gravel = skimage.img_as_ubyte(skimage.color.rgb2gray(gravel[:512, :512]))
+    >>> imageio.imwrite('gravel.png', gravel)
+    """
+    return load("gravel.png", as_gray=True)
+
```
Comment (sciunto) [inline skimage/data/__init__.py]: PEP8 missing blank line.

**Comment by sciunto** [comment]
I guess we can merge this. Stefan seems to be happy with it as well. Thanks @hmaarrfk

**Pull request merged**